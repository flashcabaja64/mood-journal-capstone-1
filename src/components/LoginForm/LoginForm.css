/* LoginForm.js containers */
.main-login {
  width: 100%;
  margin: 0 auto;
}

.wrap-login {
  display: flex;
  width: 100%;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  padding-top: 15%;
  position: relative;
  z-index: 0;
  border-radius: 27px;
}

.form-wrap-login {
  width: 60%;
  border-radius: 27px;
  overflow: hidden;
  box-shadow: 0 3px 20px 0px rgba(255, 255, 255, 0.089)
}

/* Form */
.LoginForm {
  width: 100%;
  position: relative;
  padding: 15px 0 20px 0;
}

.demo-info {
  display: block;
  line-height: 1.5;
  font-weight: 500;
  font-size: 1.3rem;
  padding-bottom: 15px;
}

/* Input fields */
.login-input {
  border-radius: 27px;
  padding: 12px 20px ;
}

input:focus {
  border-color: transparent;
}

input.login-user_name, input.login-password {
  display: block;
  position: relative;
  border-radius: 27px;
  width: 100%;
  margin-top: .5%;
  height: 35px;
  line-height: 1.2;
  text-align: center;

  font-size: 17px;
  background: #1A5486;
  color: white;
}

input::-webkit-input-placeholder { color: white;}
input:-moz-placeholder { color: white;}
input::-moz-placeholder { color: white;}
input:-ms-input-placeholder { color: white;}

/* Original [Focus Input]  */
.focus-input {
  display: block;
  position: absolute;
  z-index: -1;
  width: 100%;
  height: 100%;
  top: 0;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  -o-transform: translateX(-50%);
  transform: translateX(-50%);
  border-radius: 27px;
  background-color: rgb(22, 39, 75);
  pointer-events: none;
  
  -webkit-transition: all 0.4s;
  -o-transition: all 0.4s;
  -moz-transition: all 0.4s;
  transition: all 0.4s;
}

input:focus + .focus-input {
  width: calc(100% + 20px);
}

/* Button */

.login-btn-container {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -moz-box;
  display: -ms-flexbox;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  padding-top: 10px;
}

.login-btn {
  display: -webkit-box;
  display: -webkit-flex;
  display: -moz-box;
  display: -ms-flexbox;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 10px;
  width: 30%;
  height: 30px;
  background-color: rgb(0, 29, 54);
  border-radius: 25px;
  border: none;
  cursor: pointer;
  letter-spacing: 2px;

  font-family: 'Arial';
  line-height: 1.2;
  color: white;
  font-size: 16px;
  text-transform: uppercase;
}

button.login-btn:hover {
  background-color: #1A5486;
}

@-webkit-keyframes login-btn {
  to {
    -webkit-transform: scale(1.1);
    transform: scale(1.1);
  }
}
@keyframes login-btn {
  to {
    -webkit-transform: scale(1.1);
    transform: scale(1.1);
  }
}
.login-btn {
  display: inline-block;
  vertical-align: middle;
  -webkit-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
  box-shadow: 0 0 1px rgba(0, 0, 0, 0);
}
.login-btn:hover, .login-btn:focus, .login-btn:active {
  -webkit-animation-name: login-btn;
  animation-name: login-btn;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
}

@media only screen and (min-width: 768px) {
  .main-login, .wrap-login, .form-wrap-login,
  .LoginForm {
    width: 450px;
  }
}