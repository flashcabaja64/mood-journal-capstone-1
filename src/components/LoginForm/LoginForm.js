import React, { Component } from 'react';
import AuthApiService from '../../services/auth-api-service'
import './LoginForm.css'

export default class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  }

  state = { error: null }

  handleSubmitJwtAuth = e => {
    e.preventDefault();
    this.setState({ error: null })
    const { user_name, password } = e.target;
    console.log('submitted')

    AuthApiService.postLogin({
      user_name: user_name.value,
      password: password.value,
    })
      .then(res => {
        user_name.value = '';
        password.value = '';
        this.props.onLoginSuccess(res.authToken)
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render() {
    const { error } = this.state;
    return(
      <div className="main-login">
        <div className="wrap-login">
          <div className="form-wrap-login">
            <form
              className='LoginForm'
              onSubmit={this.handleSubmitJwtAuth}
            >
            <div role='alert'>
              {error && <p className='red'>{error}</p>}
            </div>
            <div className='login-input user_name'>
              <label id="loginUser" htmlFor='LoginForm_user_name'> Username
              </label>
              <input
                required
                className='login-user_name'
                placeholder="Username"
                name='user_name'
                id='LoginForm_user_name'
                aria-label="enter user name"
                aria-required="true"
                aria-describedby="loginUser"
              />
              <span className="focus-input"></span>
            </div>
            <div className='login-input password'>
              <label id="loginPass" htmlFor='LoginForm_password'>Password
              </label>
              <input
                required
                className='login-password'
                placeholder="Password"
                name='password'
                type='password'
                id='LoginForm_password'
                aria-label="enter user password"
                aria-required="true"
                aria-describedby="loginPass"
              />
              <span className="focus-input"></span>
            </div>
            <div className="login-btn-container">
            <button type='submit' className="login-btn">
              Login
            </button>
            </div>
            </form>
          </div>
        </div>
      </div>
    )
  }
}