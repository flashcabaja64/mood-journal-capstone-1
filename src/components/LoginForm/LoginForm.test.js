import React from 'react';
import ReactDOM from 'react-dom';
import LoginForm from './LoginForm';
import {MemoryRouter} from 'react-router';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json' 
import { shallow } from 'enzyme'

describe('LoginForm component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
  
    ReactDOM.render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>
    , div);
  
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><LoginForm /></MemoryRouter>)
        .toJSON();
        expect(tree).toMatchSnapshot();
  })
  
  it('submits information after clicking click', () => {
    const wrapper = shallow(<LoginForm />)
    wrapper.find('button').at(0).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})