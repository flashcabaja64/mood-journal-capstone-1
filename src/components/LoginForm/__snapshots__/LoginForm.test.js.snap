// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component testing renders UI as expected 1`] = `
<div
  className="main-login"
>
  <div
    className="wrap-login"
  >
    <div
      className="form-wrap-login"
    >
      <form
        className="LoginForm"
        onSubmit={[Function]}
      >
        <div
          role="alert"
        />
        <div
          className="login-input user_name"
        >
          <label
            htmlFor="LoginForm_user_name"
            id="loginUser"
          >
             Username
          </label>
          <input
            aria-describedby="loginUser"
            aria-label="enter user name"
            aria-required="true"
            className="login-user_name"
            id="LoginForm_user_name"
            name="user_name"
            placeholder="Username"
            required={true}
          />
          <span
            className="focus-input"
          />
        </div>
        <div
          className="login-input password"
        >
          <label
            htmlFor="LoginForm_password"
            id="loginPass"
          >
            Password
          </label>
          <input
            aria-describedby="loginPass"
            aria-label="enter user password"
            aria-required="true"
            className="login-password"
            id="LoginForm_password"
            name="password"
            placeholder="Password"
            required={true}
            type="password"
          />
          <span
            className="focus-input"
          />
        </div>
        <div
          className="login-btn-container"
        >
          <button
            className="login-btn"
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`LoginForm component testing submits information after clicking click 1`] = `
<div
  className="main-login"
>
  <div
    className="wrap-login"
  >
    <div
      className="form-wrap-login"
    >
      <form
        className="LoginForm"
        onSubmit={[Function]}
      >
        <div
          role="alert"
        />
        <div
          className="login-input user_name"
        >
          <label
            htmlFor="LoginForm_user_name"
            id="loginUser"
          >
             Username
          </label>
          <input
            aria-describedby="loginUser"
            aria-label="enter user name"
            aria-required="true"
            className="login-user_name"
            id="LoginForm_user_name"
            name="user_name"
            placeholder="Username"
            required={true}
          />
          <span
            className="focus-input"
          />
        </div>
        <div
          className="login-input password"
        >
          <label
            htmlFor="LoginForm_password"
            id="loginPass"
          >
            Password
          </label>
          <input
            aria-describedby="loginPass"
            aria-label="enter user password"
            aria-required="true"
            className="login-password"
            id="LoginForm_password"
            name="password"
            placeholder="Password"
            required={true}
            type="password"
          />
          <span
            className="focus-input"
          />
        </div>
        <div
          className="login-btn-container"
        >
          <button
            className="login-btn"
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
